app/exercises/five.py

```diff
@@ -65,41 +71,112 @@ class CustomToolbar(NavigationToolbar2Tk):
+        for layer, vertices in self.layers.items():
+            layers_string += f'{layer}: {vertices}\n'
+
+        self.append_text(layers_string)
 
     def generate_network(self):
         try:
             n = int(self.vertices_entry.get())
         except ValueError:
-            messagebox.showinfo(title='Wykrzyknik!', message='Wprowadź prawidłowe dane wejściowe!')
+            messagebox.showinfo(title='Wykrzyknik!',
+                                message='Wprowadź prawidłowe dane wejściowe!')
             return
 
         if n < 2:
-            messagebox.showinfo(title='Wykrzyknik!', message='Liczba wierzchołków nie może być mniejsza niż 2!')
+            messagebox.showinfo(
+                title='Wykrzyknik!', message='Liczba wierzchołków nie może być mniejsza niż 2!')
             return
 
         self.graph = WeightedDirectedAdjacencyList.init_empty()
-        layers = self.graph.generate_flow_network(n)
+        self.layers = self.graph.generate_flow_network(n)
 
         self.visualization = self.graph.convert_to_networkX()
         self.axis.clear()
-        
-        labels = nx.get_edge_attributes(self.visualization,'weight')
-        pos=nx.spring_layout(self.visualization)
-        nx.draw_networkx(self.visualization, pos=pos, ax=self.axis)
-        nx.draw_networkx_edge_labels(self.visualization, pos=pos, edge_labels=labels, ax=self.axis)
+
+        labels = nx.get_edge_attributes(self.visualization, 'weight')
+        self.pos = nx.spring_layout(self.visualization)
+        nx.draw_networkx(self.visualization, pos=self.pos, ax=self.axis)
+        nx.draw_networkx_edge_labels(
+            self.visualization, pos=self.pos, edge_labels=labels, ax=self.axis)
 
         self.canvas.draw()
         self.print_graph()
+        self.append_layers_info()
+
+    def BFS(self, s, t, path, matrix):
+
+        visited = [False for _ in range(len(matrix))]
+        visited[s] = True
+
+        Q = cs.deque([])
+        Q.append(s)
+
+        while Q:
+            u = Q.popleft()
+
+            for ind, val in enumerate(matrix[u]):
+                if visited[ind] == False and val > 0:
+                    Q.append(ind)
+                    visited[ind] = True
+                    path[ind] = u
+
+        return True if visited[t] else False
+
+    def set_flow_labels(self, flow_matrix):
+        weights = nx.get_edge_attributes(self.visualization, 'weight')
+        for i, j in weights.keys():
+            weights[(i, j)] = str(flow_matrix[j][i]) + '/' + str(weights[(i, j)])
+
+        nx.draw_networkx_edge_labels(
+            self.visualization, pos=self.pos, edge_labels=weights, ax=self.axis)
+        self.canvas.draw()
+
+    def FordFulkenson(self):
+        s = 0
+        t = len(self.graph.get_vertices())-1
+
+        matrix = self.graph.convert_to_matrix()
+        licznik = 0
+
+        path = [-1 for _ in range(len(self.graph.get_vertices()))]
+        maxFlow = 0
+
+        while self.BFS(s, t, path, matrix):
```

Comment:
As you can see, this BFS is a method and with upper case it looks like call of Functor (object with behaviour of function)